ui/apps/platform/src/services/ClustersService.ts

```diff
@@ -114,7 +115,7 @@ export function getAutoUpgradeConfig(): Promise<AutoUpgradeConfig> {
  * Saves the cluster autoupgrade config.
  */
 export function saveAutoUpgradeConfig(config: AutoUpgradeConfig): Promise<AutoUpgradeConfig> {
```

Comment:
2 problems in original frontend service function are worthwhile to fix, although unrelated to your changes.

1. The return type is inconsistent with backend service proto:

    ```proto
    rpc UpdateSensorUpgradeConfig(UpdateSensorUpgradeConfigRequest) returns (Empty) {
    ```

    Replace `Promise<AutoUpgradeConfig>` with `Promise<Empty>` a type alias we have defined.

2. The return value leaks the entire response object. Make it consistent with correct return type:

    ```ts
    return axios.post(autoUpgradeConfigUrl, wrappedObject).then((response) => response.data);
    ```

Reply:
I don't understand (2). If we change it to return `Promise<Empty>` why would we even return something?

Reply:
Excellent question. The original code returns the entire `response` object:

https://axios-http.com/docs/res_schema

The recommended code returns just the `data` property of the response which in this case is empty object `{}` which corresponds to `Promise<Empty>` return type.

Reply:
In this case, calling code has only `catch` block for error and no `then` block for response, as it should not:

```ts
        saveAutoUpgradeConfig(newConfig).catch(() => {
            // reverse the optimistic update of the control in the UI
            const rollbackConfig = {
                ...autoUpgradeConfig,
                enableAutoUpgrade: previousValue,
            };
            setAutoUpgradeConfig(rollbackConfig);

            // also, re-fetch the data from the server, just in case it did update but we didn't get the network response
            fetchConfig();
        });
```

By the way, `rollbackConfig` explains the behavior you noticed when request has 500 status.

1. First change helps TypeScript compiler report error if calling code accesses properties of response data.
2. Second change actually returns empty object which has no properties, so value is consistent with type.

Reply:
Another way to think about it: `then((response) => response.data)` is common idiom when response to POST or PUT request returns the created or updated data. It is just as necessary when response data is empty object.